performance/SwiftUIPerformanceQuickstart/PerformanceExample/Shared/Process.swift

```diff
@@ -0,0 +1,217 @@
+//
+//  Copyright 2021 Google LLC
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import FirebasePerformance
+import Vision
+
+class Process: ObservableObject {
+  @Published var status: ProcessStatus = .idle
+  @Published var action: ProcessAction?
+  @Published var image: UIImage?
+  var categories: [(category: String, confidence: VNConfidence)] = []
+  let precision: Float = 0.7
+  let trace = Performance.sharedInstance().trace(name: "Classification")
+  let site = "https://firebase.google.com/downloads/brand-guidelines/PNG/logo-logomark.png"
+  #if os(tvOS)
+    let platform = "tvOS"
+  #elseif os(iOS)
+    let platform = "iOS"
+  #endif
+
+  init() {
+    trace?.setValue("\(precision)", forAttribute: "precision")
+    trace?.setValue(platform, forAttribute: "platform")
+  }
+
+  #if swift(>=5.5)
+    @MainActor @available(iOS 15, tvOS 15,*) func updateStatusAsync(to newStatus: ProcessStatus) {
+      status = newStatus
+    }
+
+    @MainActor @available(iOS 15, tvOS 15, *) func updateImageAsync(to newImage: UIImage?) {
+      image = newImage
+    }
+
+    @MainActor @available(iOS 15, tvOS 15, *) func updateActionAsync(to newAction: ProcessAction?) {
+      action = newAction
+    }
+
+    @available(iOS 15, tvOS 15, *) func downloadImageAsync() async {
+      await updateStatusAsync(to: .running)
+
+      guard let url = URL(string: site)
+      else {
+        await updateStatusAsync(to: .failure)
+        print("Failure obtaining URL.")
+        return
+      }
+
+      do {
+        let (data, response) = try await URLSession.shared.data(from: url)
+
+        guard let httpResponse = response as? HTTPURLResponse,
+          (200 ... 299).contains(httpResponse.statusCode) else {
+          print("Did not receive acceptable response: \(response)")
+          await updateStatusAsync(to: .failure)
+          return
+        }
```

Comment:
nit: print the error if the http response isn't a 2xx code.

Reply:
I'm printing the response since I don't believe I have access to an error in that scope. My understanding is that I only have access to an error if the `try await` function call throws an error.